airbyte-integrations/connectors/source-asana/source_asana/streams.py

```diff
@@ -2,23 +2,41 @@
 # Copyright (c) 2022 Airbyte, Inc., all rights reserved.
 #
 
-from __future__ import annotations
 
 from abc import ABC
-from typing import Any, Iterable, List, Mapping, MutableMapping, Optional, Type
+from typing import Any, Iterable, Mapping, MutableMapping, Optional, Type
 
 import requests
 from airbyte_cdk.models import SyncMode
 from airbyte_cdk.sources.streams.http import HttpStream
 
+ASANA_ERRORS_MAPPING = {
+    402: "Stream is available to premium organizations and workspaces",
+    403: "Not enough permissions",
+    404: "Object specified by the request does not exist",
+    451: "This request was blocked for legal reasons",
+}
+
 
 class AsanaStream(HttpStream, ABC):
     url_base = "https://app.asana.com/api/1.0/"
-
     primary_key = "gid"
-
     # Asana pagination could be from 1 to 100.
     page_size = 100
+    raise_on_http_errors = True
+
+    @property
+    def AsanaStreamType(self) -> Type:
+        return self.__class__
+
+    def should_retry(self, response: requests.Response) -> bool:
+        if response.status_code in ASANA_ERRORS_MAPPING.keys():
+            self.logger.error(
+                f"Skipping stream {self.name}. {ASANA_ERRORS_MAPPING.get(response.status_code)}. Full error message: {response.text}"
```

Comment:
cc @erohmensing this connector is implementing a custom approach at skipping unavailable stream.

Reply:
Will add it to the list 👀

Reply:
@YowanR:

> because of how hard it was to get OAuth for this account...

Just to clarify, is enabling oAuth per-account in Asana?  Do you pay for it?